File: app/src/main/java/ar/com/sebas/lector/helper/barcode/ScannerCamera.java
package ar.com.sebas.lector.helper.barcode;

import android.content.Context;
import android.content.Intent;
import com.google.android.gms.vision.barcode.Barcode;
import ar.com.sebas.lector.helper.barcode.camera.BarcodeActivity;
import ar.com.sebas.lector.helper.barcode.camera.BarcodeFragment;
import ar.com.sebas.lector.helper.barcode.camera.BarcodeUpdateListener;

/**
 * Created by lopez.sebastian on 18/10/2017.
 *
 * Lector de códigos utilizando Mobile Vision de Google Play Services
 */

public class ScannerCamera extends AbstractScanner {
    public ScannerCamera(Context context) {
        super(context);
        BarcodeFragment.setListener(new BarcodeUpdateListener() {
            @Override
            public void onBarcodeDetected(Barcode barcode) {
                scannerListener.onRead(barcode.rawValue);
            }
        });
    }

    @Override
    public void Read() {
        Intent intentBarcode = new Intent(context, BarcodeActivity.class);
        context.startActivity(intentBarcode);
    }

    @Override
    public void Stop() {
        BarcodeActivity.Stop();
    }
}

File: app/src/main/java/ar/com/sebas/lector/helper/barcode/ScannerCamera.kt
package ar.com.sebas.lector.helper.barcode

import android.content.Context
import android.content.Intent
import ar.com.sebas.lector.helper.barcode.camera.BarcodeActivity
import ar.com.sebas.lector.helper.barcode.camera.BarcodeFragment
import ar.com.sebas.lector.helper.barcode.camera.BarcodeUpdateListener
import com.google.android.gms.vision.barcode.Barcode

/**
 * Created by lopez.sebastian on 18/10/2017.
 *
 * Lector de códigos utilizando Mobile Vision de Google Play Services
 */

class ScannerCamera(context: Context) : Scanner(context) {
    init {
        BarcodeFragment.setListener(object : BarcodeUpdateListener {
            override fun onBarcodeDetected(barcode: Barcode) {
                scannerListener.onRead(barcode.rawValue)
            }
        })
        isReady = true
    }

    override fun read() {
        val intentBarcode = Intent(context, BarcodeActivity::class.java)
        context.startActivity(intentBarcode)
    }

    override fun stop() {
        BarcodeActivity.stop()
    }
}
